include/swift/AST/LocalizationFormat.h

```diff
@@ -43,7 +131,29 @@ class LocalizationProducer {
 class YAMLLocalizationProducer final : public LocalizationProducer {
 public:
   std::vector<std::string> diagnostics;
-  explicit YAMLLocalizationProducer(std::string locale, std::string path);
+  explicit YAMLLocalizationProducer(llvm::StringRef filePath);
+  llvm::StringRef getMessageOr(swift::DiagID id,
+                               llvm::StringRef defaultMessage) const override;
+};
+
+class SerializedLocalizationProducer final : public LocalizationProducer {
+  using SerializedLocalizationTable =
+      llvm::OnDiskIterableChainedHashTable<LocalizationLookupInfo>;
+  llvm::ErrorOr<std::unique_ptr<llvm::MemoryBuffer>> Buffer;
```

Comment:
We need to make sure that it's not an error but indeed a `MemoryBuffer` because if it's an error it has to be reported and localization couldn't be used. So the field should be `std::unique_ptr<llvm::MemoryBuffer>`.